common/interpreter/error.go

```diff
@@ -2,32 +2,41 @@ package interpreter
 
 import (
 	"errors"
-	"fmt"
 	"mime"
 	"net/http"
 
 	"github.com/amp-labs/connectors/common"
 )
 
-var ErrMissingContentType = errors.New("mime.ParseMediaType failed")
+var ErrUnparseableHTTPResponse = errors.New("unparseable HTTP response")
 
 // FaultyResponseHandler used to parse erroneous response.
 type FaultyResponseHandler interface {
 	HandleErrorResponse(res *http.Response, body []byte) error
 }
 
 // ErrorHandler invokes a function that matches response media type with parse error, ex: JSON<->JsonParserMethod
-// otherwise defaults to general error interpretation.
+// otherwise defaults to general opaque error interpretation.
+// UnknownMedia is a special handler that is used in case Content-Type is unavailable.
 type ErrorHandler struct {
-	JSON FaultyResponseHandler
-	XML  FaultyResponseHandler
-	HTML FaultyResponseHandler
+	JSON         FaultyResponseHandler
+	XML          FaultyResponseHandler
+	HTML         FaultyResponseHandler
+	UnknownMedia FaultyResponseHandler
 }
 
-func (h ErrorHandler) Handle(res *http.Response, body []byte) error {
+func (h ErrorHandler) Handle(res *http.Response, body []byte) error { // nolint:cyclop
```

Comment:
I think we can rewrite this whole function like below


```
func (h ErrorHandler) Handle(res *http.Response, body []byte) error {
	mediaType, _, err := mime.ParseMediaType(res.Header.Get("Content-Type"))
	if err == nil {
		if h.JSON != nil && mediaType == "application/json" {
			return h.JSON.HandleErrorResponse(res, body)
		}

		if h.XML != nil && (mediaType == "text/xml" || mediaType == "application/xml") {
			return h.XML.HandleErrorResponse(res, body)
		}

		if h.HTML != nil && (mediaType == "text/html" || mediaType == "application/html") {
			return h.HTML.HandleErrorResponse(res, body)
		}
	}

	return common.InterpretError(res, body)
}

```

Reply:
I will do that, but I thought you wanted to treat unknown media type error handling in a special way, as per thread above: 
```
return common.HandleUnparseableHTTPResponseErrror(res, body) // return type is common.UnparseableHTTPResponseError 
```

Reply:
Initially, I thought so. But I looked at the entire function and common.InterpretError is independent of the mime type while XML, JSON, and HTML are just some special (or common) ones that we care. The goal of the initial task was to correct the error message when we are incorrectly parsing responses. `common.InterpretError` just returns a status code and returns string message from providers, that should be sufficient I think.

Reply:
That makes sense. I merging PR.